src/libpriv/rpmostree-origin.c

```diff
@@ -291,6 +300,28 @@ rpmostree_origin_set_rebase (RpmOstreeOrigin *origin,
   return TRUE;
 }
 
+/* Like g_key_file_set_string(), but remove the key if @value is NULL */
+static void
+set_or_unset_str (GKeyFile *kf,
+                  const char *group,
+                  const char *key,
+                  const char *value)
+{
+  if (!key)
+    (void) g_key_file_remove_key (kf, group, key, NULL);
+  else
+    (void) g_key_file_set_string (kf, group, key, value);
+}
+
+void
+rpmostree_origin_set_live_state (RpmOstreeOrigin *origin,
+                                 const char      *inprogress,
+                                 const char      *live)
+{
+  set_or_unset_str (origin->kf, "rpmostree-ex", "live-inprogress", inprogress);
+  set_or_unset_str (origin->kf, "rpmostree-ex", "live", live);
```

Comment:
Good catch!  Yes, fixing this also revealed another bug.  Extended the test suite for this.